File: Models/DarkSoulsData.cs
using DarkSoulsOBSOverlay.Models.Mappings;
using System.Collections.Generic;

namespace DarkSoulsOBSOverlay.Models
{
    public class DarkSoulsData
    {
        public Settings Settings { get; set; } = new();
        public bool Connected { get; set; } = false;
        public bool Loaded { get; set; } = false;
        public string Version { get; set; } = null;
        public List<KeyValuePair<int, int>> UpdatedEventFlags { get; set; } = new();
        public CharData Char { get; set; } = new();
        public EventFlags<bool> Events { get; set; } = new();
    }

    public class CharData
    {
        public string Area { get; set; } = null;
        public int AreaId { get; set; } = 0;
        public string CharacterName { get; set; } = null;
        public string CharacterClass { get; set; } = null;
        public int CharacterClassId { get; set; } = 0;
        public string LastBonfire { get; set; } = null;
        public int LastBonfireId { get; set; } = 0;
        public int Health { get; set; } = 0;
        public int HealthMax { get; set; } = 0;
        public int Deaths { get; set; } = 0;
        public double Clock { get; set; } = 0;
        public int SaveSlot { get; set; } = 0;
    }
}

File: Frontend/src/Models/DarkSoulsData.ts
import Settings from "./Settings";

export default interface DarkSoulsData {
    Settings: Settings;
    Connected: boolean;
    Loaded: boolean;
    Version: string;
    UpdatedEventFlags: number[];
    Char: CharData;
    
    //EventFlags
    Common: boolean[];
    AnorLondo: boolean[];
    Blighttown: boolean[];
    Catacombs: boolean[];
    ChasmOfTheAbyss: boolean[];
    CrystalCave: boolean[];
    DarkrootGarden: boolean[];
    DemonRuins: boolean[];
    Depths: boolean[];
    FirelinkShrine: boolean[];
    KilnOfTheFirstFlame: boolean[];
    LostIzalith: boolean[];
    NewLondoRuins: boolean[];
    OolacileTownship: boolean[];
    PaintedWorld: boolean[];
    QueelagsDomain: boolean[];
    RoyalWoods: boolean[];
    SanctuaryGarden: boolean[];
    SensFortress: boolean[];
    TombOfGiants: boolean[];
    UndeadAsylum: boolean[];
    UndeadBurg: boolean[];
    UndeadParish: boolean[];
    ValleyOfDrakes: boolean[];
}

export interface CharData {
    Area: string;
    AreaId: number;
    CharacterName: string;
    CharacterClass: string;
    CharacterClassId: number;
    LastBonfire: string;
    LastBonfireId: number;
    Health: number;
    HealthMax: number;
    Deaths: number;
    Clock: number;
    SaveSlot: number;
}